workflows/pipe-common/pipeline/cluster/cluster.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import subprocess
+from random import getrandbits
+from threading import RLock
+from . import utils
+
+lock = RLock()
```

Comment:
Lets rather create `_lock` field in `SGECluster` class and then pass it to `create_directory` method. The usage of global variables with common names can be pretty harmful.